ZzaDashboard/ViewModel/MainViewModel.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+using Zza.Data;
+
+namespace ZzaDashboard.ViewModel
+{
+    class MainViewModel : INotifyPropertyChanged
+    {
+        private Customer selectedCustomer;
+        public Customer SelectedCustomer
+        {
+            get => this.selectedCustomer;
+            set
+            {
+                if (value == this.selectedCustomer)
+                    return;
+
+                this.selectedCustomer = value;
+                this.OnPropertyChanged(nameof(this.SelectedCustomer));
+            }
+        }
+
+        private ObservableCollection<Customer> customers;
+        public ObservableCollection<Customer> Customers
+        {
+            get => this.customers;
+            set
+            {
+                if (value == this.customers)
+                    return;
+
+                this.customers = value;
+                this.OnPropertyChanged(nameof(this.Customers));
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+        protected void OnPropertyChanged(string propertyName)
+        {
+            if (this.PropertyChanged is null)
+                return;
+
+            this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+
+        public ICommand SaveCommand { get; set; }
+
+
+        public MainViewModel()
+        {
+            if (DesignerProperties.GetIsInDesignMode(new DependencyObject()))
+                return;
+
+            this.Customers = GetAllCustomers();
+
+            this.SaveCommand = new RelayCommand(SaveCommandExecuted, SaveCommandCanExecute);
+
+        }
+
+        private ObservableCollection<Customer> GetAllCustomers()
+        {
+            string hugeText = string.Empty;
+            ObservableCollection<Customer> customers = new ObservableCollection<Customer>();
+            using (var reader = new StreamReader($@"{Directory.GetCurrentDirectory()}\ZzaPersons.txt"))
+            {
+                hugeText = reader.ReadToEnd();
+            }
+            foreach (var line in hugeText.Split(Environment.NewLine.ToCharArray()))
+            {
+                if (!string.IsNullOrEmpty(line))
```

Comment:
I'd suggest to invert if to avoid nesting.

Reply:
you mean this way?
                if (string.IsNullOrEmpty(line))
                    continue;

                var customer = new Customer();
                customer.FirstName = line.Split('~')[0];
                customer.LastName = line.Split('~')[1];
                customer.Phone = line.Split('~')[2];
                customers.Add(customer);

Reply:
Yep